pkg/apis/certmanager/types.go

```diff
@@ -0,0 +1,48 @@
+package certmanager
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient=true
+
+type Certificate struct {
+	metav1.TypeMeta
+	metav1.ObjectMeta
+
+	Spec CertificateSpec
+}
+
+type CertificateList struct {
+	metav1.TypeMeta
+	metav1.ListMeta
+
+	Items []Certificate
+}
+
+type CertificateSpec struct {
+	Domain     string
+	Provider   string
+	Email      string
+	SecretName string
+	AltNames   []string
+}
+
+type ACMECertData struct {
+	DomainName string
+	Cert       []byte
+	PrivateKey []byte
+}
+
+type ACMEUserData struct {
+	Email string
+	// Registration *acme.RegistrationResource
```

Comment:
We shouldn't be referencing external packages within our types files, as it will cause issues when converting between API versions in future.

I am thinking we'll be best to copy this structure from the ACME library, despite the duplication, so that if anything is to change in future we can provide a clean upgrade path for users.